### PR TITLE
[IGNORE] enable checking format of the markdown docs

### LINKED ISSUE #889: Set up markdown formatter for consistency
We've had a few PRs where different folks have different default formatters from their text editors auto-changing markdown, which leads to messy diffs. We should set up a consistent tool for this to avoid the noise, ideally one that easily integrates into the text editors being used by most maintainers.

I think the UI devs are mostly using vscode, and we use prettier for formatting our UI code, so that might be a good option, but I'm not sure how well that would work for our backend collaborators who are using intellij. Will need to investigate.

### PR DESCRIPTION
**Pull request by Nexucis**
This PR introduced the usage of [mdox](https://github.com/bwplotka/mdox) to ensure the different documentation written in the markdown files are well formatted.

The tool will also ensure that every links are working. Spoiler alert, some were broken :).

It fixed #889 

Signed-off-by: Augustin Husson <husson.augustin@gmail.com>

### CODE REVIEW

**Review by AntoineThebaud — APPROVED** [review]

**Comment by juliepagano** [comment]
Some additional adjustments are likely needed to avoid prettier formatting markdown in the `ui` directory. Adding something like `*.md` to `ui/.prettierignore` should do it.

**Comment by juliepagano** [comment]
The one down side of this approach from my perspective as a UI dev is that it takes formatting I got immediately on saving a file (i.e. automatically happens without much thought) into yet another script I might forget to run before creating a PR. 

I think that could be somewhat addressed by making a little "pre-check" script or something similar that wraps those "quick to run but easy to forget" checks into a script you could opt into running locally. Might include stuff like:
- docs formatting
- DCO check
- linting

Maybe a useful follow-up task after this merges?

**Review by cndonovan — APPROVED** [review]
Adding a tool that do these checks is nice / helps the code be better 👍 

I suspect that if, at some point in the future, I'm writing or editing a Markdown file, I will forget to run a script to do this formatting (like Julie said, as a frontend dev, I'm used to `prettier` handling formatting for me). 

So it's likely I'd forget... but we can always re-run this tool if people have forgotten after making a bunch of Markdown changes.

**Review by juliepagano — APPROVED** [review]
LGTM once the prettierignore is adjusted [see comment](https://github.com/perses/perses/pull/924#issuecomment-1386098521). Thanks for fixing this up!

File: ui/e2e/README.md
Comment (Nexucis) [reply ui/e2e/README.md:59]: nah it's me that fixed that haha.

File: Makefile
Comment (Nexucis) [reply Makefile:59]: yeah perhaps we should align the way to write makefile directive indeed.

**Comment by Nexucis** [comment]
on my side, I'm relying on IntelliJ to format the code haha. So I might forget as well to format the markdown document with that script :). 

Hopefully this formatting is not that intrusive, so probably most of the time, it won't block anything. I'm even wondering if you run the prettier formatting on markdown, then the format check on the doc will be green.

But for sure, we could imagine a script that could be added as a pre commit hook in your local git. So everytime you are creating a commit, it can format and check your commit.
Usually I don't like this kind of thing because it is slowing down the creation of a commit. But that's just my opinion, if others like that, we can always provide the way to do it.

**Pull request merged**
